homeassistant/components/trafikverket_weatherstation/config_flow.py

```diff
@@ -30,35 +26,52 @@ class TVWeatherConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
 
     entry: config_entries.ConfigEntry
 
+    async def validate_input(self, sensor_api: str, station: str):
+        """Validate input from user input."""
+        web_session = async_get_clientsession(self.hass)
+        weather_api = TrafikverketWeather(web_session, sensor_api)
+        try:
+            await weather_api.async_get_weather(station)
+        except ValueError as err:
```

Comment:
For the future I would recommend that the library should raise library specific exceptions that the library user can import and catch. Each different kind of reason should be a different exception. Relying on the exception argument of a generic `ValueError` isn't robust or convenient. Would be nice if the library can be improved in that direction.

Reply:
Agree. I didn't make it. I just looked into it to extract the various messages that comes from it